generic/build

```diff
@@ -74,7 +74,10 @@ geturl() {
 		"${WGET}" ${WGET_OPTS} --directory-prefix="${SOURCESROOT}" "${url}"
 	else
 		(
-			cd "${SOURCESROOT}" && \
+		if [ ! -d "${SOURCESROOT}" ]; then
+			mkdir -p "${SOURCESROOT}"
```

Comment:
Interesting...looking at _generic/build_ I see that the create_layout() function creates SOURCESROOT, but it runs _after_ the download() function. Have you tried reverting the position of those functions, so that the main() function would look like this?
```
main() {
        export CFLAGS="${TARGET_CFLAGS} ${EXTRA_TARGET_CFLAGS}"
        export LDFLAGS="${TARGET_LDFLAGS} ${EXTRA_TARGET_LDFLAGS}"
        create_layout
        download
        extract
        build_dep
        --- snip ---
```
Would that solve this problem?